lib/kb_hmmer/kb_hmmerImpl.py

```diff
@@ -1633,8 +1646,13 @@ def HMMER_Local_MSA_Group_Search(self, ctx, params):
                 sequence_str = seq_obj['sequence']
 
                 PROT_pattern = re.compile("^[acdefghiklmnpqrstvwyACDEFGHIKLMNPQRSTVWYxX ]+$")
-                #DNA_pattern = re.compile("^[acgtuACGTUnryNRY ]+$")
-                if not PROT_pattern.match(sequence_str):
+                DNA_pattern = re.compile("^[acgtuACGTUnryNRY ]+$")
+                if DNA_pattern.match(sequence_str):
+                    self.log(invalid_msgs,
+                             "Require protein sequences for target. " +
+                             "BAD nucleotide record for sequence_id: "+header_id+"\n"+sequence_str+"\n")
+                    continue
+                elif not PROT_pattern.match(sequence_str):
```

Comment:
This logic is repeated several times. Consider encapsulating in a function to save yourself future effort.

Reply:
I made a method out of the MSA sequence type validation, as requested.